src/utils/io.ts

```diff
@@ -0,0 +1,193 @@
+import fs from 'fs'
+import axios from 'axios'
+import AWS from 'aws-sdk'
+import {createWinstonLogger} from './logging'
+import winston from 'winston'
+
+export async function readFileURI(uri: string, encoding: BufferEncoding, logger?: winston.Logger): Promise<string> {
+  if (!logger) {
+    logger = createWinstonLogger('File IO', 'warn')
+  }
+
+  let parsedURI
+
+  try {
+    parsedURI = new URL(uri)
+    logger.debug(`Parsed URI: ${uri} - Protocol is ${parsedURI.protocol}`)
+  } catch {
+    return fs.readFileSync(uri, encoding)
```

Comment:
Is this really the best place in our 'switch' statement to get the 'I am reading a file from the filesystem' case?